types/connect/index.d.ts

```diff
@@ -1,94 +1,163 @@
-// Type definitions for connect v3.4.0
+// Type definitions for connect 3.7
 // Project: https://github.com/senchalabs/connect
 // Definitions by: Maxime LUCE <https://github.com/SomaticIT>
 //                 Evan Hahn <https://github.com/EvanHahn>
+//                 geekley <https://github.com/geekley>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
-
 import * as http from "http";
+import connect = createServer;
 
 /**
- * Create a new connect server.
- * @public
+ * When the `connect` module is required, a function is returned that will construct a new app when called.
+ * This app will store all the middleware added and is, itself, a function.
+ *
+ * The core of Connect is "using" middleware. Middleware are added as a "stack" where incoming
+ * requests will execute each middleware one-by-one until a middleware does not call `next()` within it.
+ *
+ * There are special cases of "error-handling" middleware. These are middleware where the
+ * function takes exactly 4 arguments. When a middleware passes an error to `next`, the app will
+ * proceed to look for the error middleware that was declared after that middleware and invoke it,
+ * skipping any error middleware above that middleware and any non-error middleware below.
+ *
+ * The last step is to actually use the Connect app in a server. The `.listen()` method is a
+ * convenience to start a HTTP server (and is identical to the `http.Server`'s `listen` method).
+ * The app itself is really just a function with three arguments, so it can also be handed to
+ * `.createServer()` in Node.js.
  */
-declare function createServer(): createServer.Server;
+declare function createServer(): connect.Server;
 
 declare namespace createServer {
+    export interface Request extends http.IncomingMessage {
+        /**
+         * The request URL, altered to remove the route part. For example, if the handle function is used
+         * at the route `/foo`, the request for `/foo/bar` will invoke it with `req.url === '/bar'` and
+         * `req.originalUrl === '/foo/bar'`.
+         */
+        url: string;
```

Comment:
This is [already a property of `http.IncomingMessage`](https://nodejs.org/api/http.html#http_message_url). Have you added this for documentation purposes?

Reply:
Yes, the documentation override is important because this url is conceptually different from the superclass (since it doesn't contain the route part). Also the not-null guarantee. It helps if you have strict null checking, you can do `req.url` without an assertion.